src/scanspec/core.py

```diff
@@ -423,7 +430,20 @@ def extract_gap(gaps: Iterable[GapArray]) -> GapArray | None:
                     return gap[dim_indices]
             return None
 
-        return _merge_frames(self, dict_merge=extract_dict, gap_merge=extract_gap)
+        def extract_duration(
+            durations: Iterable[DurationArray | None],
+        ) -> DurationArray | None:
+            for d in durations:
+                if d is not None:
+                    return d[dim_indices]
+                return None
```

Comment:
For consistency with the above functions

```suggestion
            return None
```